mlflow/johnsnowlabs/__init__.py

```diff
@@ -758,12 +762,24 @@ def _load_pyfunc(path, spark=None):
 
     Returns:
         None.
-
     """
-    return _PyFuncModelWrapper(
-        _load_model(model_uri=path),
-        spark or _get_or_create_sparksession(path),
-    )
+    try:
+        popped_envs = {}
+        if not is_in_databricks_runtime() or is_in_databricks_model_serving_environment():
+            # Johnsnowlab code uses incorrect logic to check if it is in Databricks Runtime
+            # if there is any environmental variable key contains 'DATABRICKS', it is regarded
+            # as the databricks runtime.
+            # to fix this, we popped these environmental variables temporarily as the patch.
+            for k in os.environ:
+                if 'DATABRICKS' in k:
+                    v = os.environ.pop(k)
+                    popped_envs[k] = v
```

Comment:
Is this really safe?

Reply:
This might cause different issues.

Reply:
I think it should be safe. Because the modified env only wraps the part of "loading model". "loading model" logic has nothing to do with databricks internal code

Reply:
Got it, I'm fine if this works.